packages/eservice-template-process/src/services/eserviceTemplateService.ts

```diff
@@ -298,6 +303,25 @@ async function parseAndCheckAttributes(
   };
 }
 
+const retrieveDocument = (
+  eserviceTemplateId: EServiceTemplateId,
+  eserviceTemplateVersion: EServiceTemplateVersion,
+  documentId: EServiceDocumentId
+): Document => {
+  const document = [
+    ...eserviceTemplateVersion.docs,
+    eserviceTemplateVersion.interface,
+  ].find((doc) => doc != null && doc.id === documentId);
+  if (document === undefined) {
+    throw eserviceTemplateDocumentNotFound(
+      eserviceTemplateId,
+      eserviceTemplateVersion.id,
+      documentId
+    );
+  }
+  return document;
+};
```

Comment:
Could you update the branch and use this function also in the `getEServiceTemplateDocument` service created by @Alepazz ?